openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -0,0 +1,162 @@
+"""
+Tests for all the views in applications app.
+"""
+
+import mock
+from django.test import TestCase, RequestFactory, Client
+from django.urls import reverse
+
+from openedx.adg.lms.applications.models import ApplicationHub
+from openedx.adg.lms.applications.views import ApplicationHubView
+from student.tests.factories import UserFactory
+from .constants import USERNAME, EMAIL, PASSWORD
+
+
+class ApplicationHubViewTest(TestCase):
+    """
+    TestCase for ApplicationHubView.
+    """
+
+    def setUp(self):
+        """
+        Setup the User and ApplicationHub for testing.
+        """
+        self.user = UserFactory(username=USERNAME, email=EMAIL, password=PASSWORD)
+        self.user_application = ApplicationHub.objects.create(user=self.user)
+        self.user_application.save()
+        factory = RequestFactory()
+        self.request = factory.get(reverse('application_hub'))
+        self.request.user = self.user
+        self.response = ApplicationHubView.as_view()(self.request)
```

Comment:
I have read the documentation and I get your point regarding this.
However, I have used the RequestFactory in order to avoid using Client. The former does not hit the template middleware to fetch the template whereas the latter does. 
Our test environment in adg has not been setup for the new theme directory, as a result, requesting any url (with a corresponding template) with Client ends up with a `TemplateDoesNotExist` error and the code breaks. 
I was told to find a workaround where I do not use the actual template and still test the functionality.

Reply:
Okay that seems a good enough reason 🙂